quipucords/api/networkprofile_serializer.py

```diff
@@ -131,14 +134,181 @@ def validate_name(name):
+            r'^{0}\.{0}\.{0}\.\[{0}:{0}\]$'.format(octet_regex),
+            r'^{0}\.{0}\.\[{0}:{0}\]\.{0}$'.format(octet_regex),
+            r'^{0}\.{0}\.\[{0}:{0}\]\.\[{0}:{0}\]$'.format(octet_regex)
+        ]
+
+        # type HOST:                abcd
+        # type HOST NUMERIC RANGE:  abcd[2:4].foo.com
+        # type HOST ALPHA RANGE:    abcd[a:f].foo.com
+        host_regex_list = [
+            r'[a-zA-Z0-9-\.]+',
+            r'[a-zA-Z0-9-\.]*\[[0-9]+:[0-9]+\]*[a-zA-Z0-9-\.]*',
+            r'[a-zA-Z0-9-\.]*\[[a-zA-Z]{1}:[a-zA-Z]{1}\][a-zA-Z0-9-\.]*']
+
+        normalized_hosts = []
+        host_errors = []
+        for host in hosts:
+            result = None
+            host_range = host['host_range']
+
+            ip_match = re.match(relaxed_ip_pattern, host_range)
+            cidr_match = re.match(relaxed_cidr_pattern, host_range)
+            is_likely_ip = ip_match and ip_match.end() == len(host_range)
+            is_likely_cidr = cidr_match and cidr_match.end() == len(host_range)
+            if is_likely_ip or is_likely_cidr:
+                # This is formatted like an IP or CIDR
+                # (e.g. #.#.#.# or #.#.#.#/#)
+                for reg in ip_regex_list:
+                    match = re.match(reg, host_range)
+                    if match and match.end() == len(host_range):
+                        result = host
+                        break
+
+                if result is None or is_likely_cidr:
+                    # Attempt to convert CIDR to ansible range
+                    if is_likely_cidr:
+                        try:
+                            normalized_cidr = NetworkProfileSerializer \
+                                .cidr_to_ansible(host_range)
+                            normalized_cidr_host = {
+                                'host_range': normalized_cidr}
+                            result = normalized_cidr_host
+                        except ValidationError as validate_error:
+                            result = validate_error
+                    else:
+                        err_message = '{} is not a valid IP or CIDR pattern'
+                        result = ValidationError(err_message
+                                                 .format(host_range))
+            else:
+                # Possibly a host addr
+                for reg in host_regex_list:
+                    match = re.match(reg, host_range)
+                    if match and match.end() == len(host_range):
+                        result = host
+                        break
+                if result is None:
+                    result = ValidationError(
+                        '{} is invalid host'.format(host_range))
+
+            if isinstance(result, ValidationError):
+                host_errors.append(result)
+            elif result is not None:
+                normalized_hosts.append(result)
+            else:
+                logging.error('%s did not match a pattern or produce error',
```

Comment:
I feel like we should be capturing some kind of unknown format or cannot process validation error here.